pep2bib.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+# usage: python3 pep2bib.py .
+
+import glob
+import os
+import re
+import sys
+from pybtex.database import Entry, BibliographyData
+
+from pep_parsing_helpers import pep_number, pep_creation_dt, first_line_starting_with, parse_authors
+
+BIB_PATH = os.path.join(sys.argv[1], 'peps.bib')
+
+
+name_first_regex = re.compile(r'(.*)<.*>')
+mail_first_regex = re.compile(r'.*\((.*)\)')
+name_only_regex = re.compile(r'(.*)')
+
+
+months = {
+    1: 'jan',
+    2: 'feb',
+    3: 'mar',
+    4: 'apr',
+    5: 'may',
+    6: 'jun',
+    7: 'jul',
+    8: 'aug',
+    9: 'sep',
+    10: 'oct',
+    11: 'nov',
+    12: 'dec',
+}
+
+
+def authors_to_bib(authors):
+    cleaned = []
+    for author in authors:
+        match = name_first_regex.match(author)
+        if match is None:
+            match = mail_first_regex.match(author)
+        if match is None:
+            match = name_only_regex.match(author)
+        cleaned.append(match.group(1).strip())
+    return " and ".join(cleaned)
+
+
+def main():
+    # get list of peps with creation time
+    # (from "Created:" string in pep .rst or .txt)
+    peps = glob.glob('pep-*.txt')
+    peps.extend(glob.glob('pep-*.rst'))
+
+    peps_with_dt = [(pep_number(full_path), pep_creation_dt(full_path), full_path) for full_path in peps]
+    # sort peps by number
+    peps_with_dt.sort()
+
+    items = {}
+    for n, dt, full_path in peps_with_dt:
+        title = first_line_starting_with(full_path, 'Title:')
+        author_string = first_line_starting_with(full_path, 'Author:')
+        authors = parse_authors(author_string)
+        authors = authors_to_bib(authors)
+        url = 'https://www.python.org/dev/peps/pep-%0.4d/' % n
```

Comment:
Might want to factor this into a top-level constant, so its easier to update if and when we move to `peps.python.org`